semcore/d3-chart/src/Pattern.tsx

```diff
@@ -0,0 +1,645 @@
+          />
+        </>
+      ),
+    },
+    symbol: {
+      viewBox: '0 0 21 20',
+      size: [16, 17.14],
+      children: (
+        <>
+          <path d='M8.823 1.336 6.596 0h8.62a8.593 8.593 0 0 1 1.29 4.512c0 3.183-1.78 6.116-4.823 7.941-1.59.954-2.125 2.19-2.125 3.176 0 .985.535 2.22 2.125 3.175L13.677 20H5.206A8.578 8.578 0 0 1 4 15.629c0-3.184 1.781-6.117 4.823-7.942 1.59-.954 2.125-2.19 2.125-3.175 0-.986-.534-2.222-2.125-3.176Z' />
+        </>
+      ),
+    },
+  },
+  squareOutline: {
+    fill: {
+      viewBox: '0 0 22 22',
+      children: (
+        <>
+          <path
+            fillRule='evenodd'
+            d='M2 2h16v16H2V2Zm4.364 4.364v7.272h7.272V6.364H6.364Z'
+            clipRule='evenodd'
+          />
+        </>
+      ),
+    },
+    symbol: {
+      viewBox: '0 0 21 20',
+      size: [18, 18],
+      children: (
+        <>
+          <path fillRule='evenodd' d='M1.5 1.5h17v17h-17v-17Zm5 5v7h7v-7h-7Z' clipRule='evenodd' />
+        </>
+      ),
+    },
+  },
+  triangle: {
+    fill: {
+      viewBox: '0 0 21 20',
+      children: (
+        <>
+          <path d='m10 2 8.66 15H1.34L10 2Z' />
+        </>
+      ),
+    },
+    symbol: {
+      viewBox: '0 0 33 32',
+      size: [16, 16],
+      children: (
+        <>
+          <path d='m15.66 5 12.66 21.928H3L15.66 5Z' />
+        </>
+      ),
+    },
+  },
+  crescent: {
+    fill: {
+      viewBox: '0 0 21 20',
+      children: (
+        <>
+          <path d='M18.501 16.748c.269-.293.018-.748-.38-.748A9 9 0 0 1 11.466.94c.32-.35.13-.941-.345-.941-5.523 0-10 4.477-10 10s4.477 10 10 10a9.974 9.974 0 0 0 7.38-3.252Z' />
+        </>
+      ),
+    },
+    symbol: {
+      viewBox: '0 0 21 20',
+      size: [14.99, 17.14],
+      children: (
+        <>
+          <path d='M18.501 16.748c.269-.293.018-.748-.38-.748A9 9 0 0 1 11.466.94c.32-.35.13-.941-.345-.941-5.523 0-10 4.477-10 10s4.477 10 10 10a9.974 9.974 0 0 0 7.38-3.252Z' />
+        </>
+      ),
+    },
+  },
+  zigzag: {
+    fill: {
+      viewBox: '0 0 12 12',
+      children: (
+        <>
+          <path d='M6 4.621 1.379 0H5.62L6 .379 6.379 0h4.242L6 4.621ZM0 10.621 1.379 12H5.62L0 6.379v4.242ZM6.379 12h4.242L12 10.621V6.38L6.379 12Z' />
+        </>
+      ),
+    },
+    symbol: {
+      viewBox: '0 0 21 21',
+      size: [17.14, 16.02],
+      children: (
+        <>
+          <path d='m20.121 1.284-9.797 9.797L.121.88v8.485l10.203 10.203 9.797-9.797V1.284Z' />
+        </>
+      ),
+    },
+  },
+} satisfies Record<string, Pattern>;
+
+const defaultPatternsList = Object.values(defaultPatterns);
+const patternsSync = new WeakMap();
+patternsSync.set(defaultPatternsList, []);
+
+export type PatternsConfig = true | keyof typeof defaultPatterns | Pattern | Pattern[];
+
+const getPatternByKey = (patternKey: string, patternsConfig: PatternsConfig = true) => {
```

Comment:
let's add here return type. Am I right, that it should be Pattern?

Reply:
Yes, you are right